src/modules/Favorites/components/FavoritesAlert.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { View, StyleSheet, Text, TouchableOpacity, Modal } from 'react-native';
+
+import { colors } from '../../../constants/colors';
+
+export const FavoritesAlert = ({
+  onPressActionButton,
+  checkFavoriteStatus,
+  selectedTrackId,
+  setSelectedTrackId,
+}) => {
+  const actionWord = checkFavoriteStatus() ? 'Remove' : 'Add';
+
+  const onAction = () => onPressActionButton(checkFavoriteStatus());
+  const onCancel = () => setSelectedTrackId('');
```

Comment:
I suggest removing these functions from `FavoritesAlert` component. Please, pass `onCloseModal` and `onSubmit` to the `FavoritesAlert` component. All logic should be located in right place